h2o-core/src/test/java/water/rapids/RapidsTest.java

```diff
@@ -524,4 +529,50 @@ static boolean checkSaneFrame_impl() {
     }
   }
 
+ @Test public void testCumu() {
+    Frame fr = null,fr2=null;
+    String x = null;
+    try {
+      fr = ArrayUtils.frame("c1",vec(ari(1,1)));
+      fr.    add("c2"  ,vec(ari(2,2)));
+      fr.    add("c3"  ,vec(ari(3,3)));
+      fr.    add("c4"  ,vec(ari(4,4)));
+      fr = new Frame(fr);
+      DKV.put(fr);
+      x = String.format("(cumsum %s 1)", fr._key);
+      Val val = Rapids.exec(x);
+      if (val instanceof ValFrame ) {
```

Comment:
You probably want to have an assert here: if val is not a frame, then it must be an error...